src/Datadog.Trace/DuckTyping/DuckType.Methods.cs

```diff
@@ -308,7 +315,35 @@ private static void CreateMethods(TypeBuilder proxyTypeBuilder, Type proxyType,
                             }
 
                             // If the target parameter type is public or if it's by ref we have to actually use the original target type.
-                            targetParamType = UseDirectAccessTo(targetParamType) ? targetParamType : typeof(object);
+                            targetParamType = UseDirectAccessTo(proxyTypeBuilder, targetParamType) ? targetParamType : typeof(object);
+                            il.WriteSafeTypeConversion(proxyParamType, targetParamType);
+
+                            targetMethodParametersTypes[idx] = targetParamType;
+                        }
+                        else
+                        {
+                            if (NeedsDuckChaining(proxyParamType, targetParamType))
```

Comment:
Now that we're creating Duck types both ways, I think it would help readability if we switched the order of the arguments so that the first argument is the "From" type and the second argument is the "To" type and if we also used the parameter names in this invocation specifically to make clear that this is going the opposite way. I think this feedback could be applied in a separate refactoring PR so it doesn't confuse reviewers on the real changes needed for reverse proxying

Reply:
Totally agree. I'll do that in another PR once this one gets merged.